diff --git a/boost_1_80_0/libs/locale/build/Jamfile.v2 b/boost_1_80_0/libs/locale/build/Jamfile.v2
index 450ec87acd..180251501a 100644
--- a/boost_1_80_0/libs/locale/build/Jamfile.v2
+++ b/boost_1_80_0/libs/locale/build/Jamfile.v2
@@ -79,11 +79,21 @@ ICU_LINK = [ modules.peek : ICU_LINK ] ;
 # Note that ICU_LINK and ICU_LINK_LOCALE will be removed in Boost 1.81
 # in favor of a similar solution to Boost.Regex or a B2 module.
 ICU_LINK ?= [ modules.peek : ICU_LINK_LOCALE ] ;
+ICU_STATIC_RUNTIME =  [ modules.peek : ICU_STATIC_RUNTIME ] ;
 
 if $(ICU_LINK)
 {
-    ICU_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin <runtime-link>shared ;
-    ICU64_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin64 <runtime-link>shared ;
+    if $(ICU_STATIC_RUNTIME)
+    {
+        ICU_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin <runtime-link>static ;
+        ICU64_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin64 <runtime-link>static ;
+    }
+    else
+    {
+        ICU_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin <runtime-link>shared ;
+        ICU64_OPTS = <include>$(icu-path)/include <linkflags>$(ICU_LINK) <dll-path>$(icu-path)/bin64 <runtime-link>shared ;
+    }
+    
 } else
 {
     searched-lib icuuc : :  <name>icuuc
