--- a/boost_1_72_0/boost/system/detail/system_category_win32.hpp
+++ b/boost_1_72_0/boost/system/detail/system_category_win32.hpp
@@ -95,7 +95,7 @@
             FORMAT_MESSAGE_FROM_SYSTEM_ | FORMAT_MESSAGE_IGNORE_INSERTS_,
             NULL,
             ev,
-            MAKELANGID_( LANG_NEUTRAL_, SUBLANG_DEFAULT_ ), // Default language
+            MAKELANGID_( LANG_ENGLISH_, SUBLANG_ENGLISH_US_ ), // Default language
             buffer,
             static_cast<DWORD_>( len ),
             NULL
@@ -116,7 +116,7 @@
             FORMAT_MESSAGE_ALLOCATE_BUFFER_ | FORMAT_MESSAGE_FROM_SYSTEM_ | FORMAT_MESSAGE_IGNORE_INSERTS_,
             NULL,
             ev,
-            MAKELANGID_( LANG_NEUTRAL_, SUBLANG_DEFAULT_ ), // Default language
+            MAKELANGID_( LANG_ENGLISH_, SUBLANG_ENGLISH_US_ ), // Default language
             (LPWSTR_) &lpMsgBuf,
             0,
             NULL
@@ -174,7 +174,7 @@
         FORMAT_MESSAGE_ALLOCATE_BUFFER_ | FORMAT_MESSAGE_FROM_SYSTEM_ | FORMAT_MESSAGE_IGNORE_INSERTS_,
         NULL,
         ev,
-        MAKELANGID_( LANG_NEUTRAL_, SUBLANG_DEFAULT_ ), // Default language
+        MAKELANGID_( LANG_ENGLISH_, SUBLANG_ENGLISH_US_ ), // Default language
         (LPWSTR_) &lpMsgBuf,
         0,
         NULL
--- a/boost_1_72_0/boost/winapi/error_handling.hpp
+++ b/boost_1_72_0/boost/winapi/error_handling.hpp
@@ -67,8 +67,10 @@
 
 BOOST_CONSTEXPR_OR_CONST WORD_ LANG_NEUTRAL_=                  LANG_NEUTRAL;
 BOOST_CONSTEXPR_OR_CONST WORD_ LANG_INVARIANT_=                LANG_INVARIANT;
+BOOST_CONSTEXPR_OR_CONST WORD_ LANG_ENGLISH_=                  LANG_ENGLISH;
 
 BOOST_CONSTEXPR_OR_CONST WORD_ SUBLANG_DEFAULT_=               SUBLANG_DEFAULT;    // user default
+BOOST_CONSTEXPR_OR_CONST WORD_ SUBLANG_ENGLISH_US_=            SUBLANG_ENGLISH_US;
 
 BOOST_FORCEINLINE BOOST_CONSTEXPR WORD_ MAKELANGID_(WORD_ p, WORD_ s) BOOST_NOEXCEPT
 {
@@ -96,8 +98,10 @@
 
 BOOST_CONSTEXPR_OR_CONST WORD_ LANG_NEUTRAL_=                  0x00;
 BOOST_CONSTEXPR_OR_CONST WORD_ LANG_INVARIANT_=                0x7f;
+BOOST_CONSTEXPR_OR_CONST WORD_ LANG_ENGLISH_=                  0x09;
 
 BOOST_CONSTEXPR_OR_CONST WORD_ SUBLANG_DEFAULT_=               0x01;    // user default
+BOOST_CONSTEXPR_OR_CONST WORD_ SUBLANG_ENGLISH_US_=            0x01;
 
 BOOST_FORCEINLINE BOOST_CONSTEXPR WORD_ MAKELANGID_(WORD_ p, WORD_ s) BOOST_NOEXCEPT
 {
